### ISSUE TITLE
OpenShift takes too long to show catalog (templates) items

### ISSUE THREAD

**Issue by davivcgarcia** [openshift/origin#16485]
##### Description
When the internet connection is poor, OpenShift takes too long to show catalog items.

##### Version
```
[root@ocp-master-1 ~]# oc version 
oc v3.5.5.31.24
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO
Server https://console...:8443
openshift v3.5.5.31.24
kubernetes v1.5.2+43a9be4
```

```
[root@ocp-master-1 ~]# openshift version
openshift v3.5.5.31.24
kubernetes v1.5.2+43a9be4
etcd 3.1.0
```

##### Steps To Reproduce
1. Using Google Chrome with Developer Tool (Ctrl+I), open OpenShift Console.
2. Create a new project, or click on 'Add to Project'
3. Verify on 'Network' tab the time consumed to download the 'templates' file.

##### Current Result
It takes from 5 seconds up to 10 seconds.

##### Expected Result
It should load faster (maybe 2 seconds).

##### Technical Details
Looks like there is a asynchronous JavaScript request to the backend API (`/oapi/v1/namespaces/openshift/templates`) in order to collect the list of templates. However, the API returns the JSON list in a human readable format, consuming by default around 3.2MB.

IMHO, we should minify this response in order to save bandwidth and improve UI response time. I tested with `json-minify` and I was able to reduce its size in around 60%, to around 1.4MB.

**Comment by spadgett** [comment]
@davivcgarcia Thanks for the report. We've fixed this recently in https://github.com/openshift/origin-web-console/pull/1919

**Comment by davivcgarcia** [comment]
Is it going to be fixed on v3.5?

**Comment by spadgett** [comment]
Unfortunately, the API to request only metadata is new in 3.7, so it's a difficult fix to backport.
